src/AbstractArray.php

```diff
@@ -180,7 +180,7 @@ public function __construct(array $elements = [])
      *
      * @return static The instance array with combined elements
      */
-    abstract public function combineTo(array $array);
+    abstract public function useAndCombineTo(array $array);
```

Comment:
Yes, you understand it right. The only reason to add `useAndReplaceIn` is performance. With it you don't need to create a second object to do this operation, you work only with the first one (In case to use `MutableArray`).